src/services/namingService.test.ts

```diff
@@ -86,4 +86,60 @@ describe("Naming Service", () => {
     const actual = AzureNamingService.getNormalizedRegionName(expected);
     expect(actual).toEqual(expected);
   });
+
+  it("deployment name is generated correctly", () => {
+    const config: ServerlessAzureConfig = {
+      functions: [],
+      plugins: [],
+      provider: {
+        prefix: "sls",
+        name: "azure",
+        region: "westus",
+        stage: "dev",
+      },
+      service: "test-api"
+    };
+
+    const timestamp = Date.now().toString();
+
+    const deploymentName = AzureNamingService.getDeploymentName(config, `t${timestamp}`);
+    expect(deploymentName.length).toBeLessThanOrEqual(64);
+    expect(deploymentName).toContain(timestamp);
+    expect(deploymentName).toContain(AzureNamingService.createShortAzureRegionName(config.provider.region));
+    expect(deploymentName).toContain(createSafeString(config.provider.prefix));
+    expect(deploymentName).toContain(createSafeString(config.provider.stage));
+    expect(deploymentName).toContain(createSafeString(config.service));
```

Comment:
also add a test for the right order <prefix-stage-region-service-timestamp>?

Reply:
Added more explicit expectation for each test case to validate full expected name.